// @aws-sdk/client-ecs/dist-es/models/models_0.js
import { __assign, __extends } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { ECSServiceException as __BaseException } from "./ECSServiceException";
var AccessDeniedException = (function (_super) {
    __extends(AccessDeniedException, _super);
    function AccessDeniedException(opts) {
        var _this = _super.call(this, __assign({ name: "AccessDeniedException", $fault: "client" }, opts)) || this;
        _this.name = "AccessDeniedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AccessDeniedException.prototype);
        return _this;
    }
    return AccessDeniedException;
}(__BaseException));
export { AccessDeniedException };
export var AgentUpdateStatus;
(function (AgentUpdateStatus) {
    AgentUpdateStatus["FAILED"] = "FAILED";
    AgentUpdateStatus["PENDING"] = "PENDING";
    AgentUpdateStatus["STAGED"] = "STAGED";
    AgentUpdateStatus["STAGING"] = "STAGING";
    AgentUpdateStatus["UPDATED"] = "UPDATED";
    AgentUpdateStatus["UPDATING"] = "UPDATING";
})(AgentUpdateStatus || (AgentUpdateStatus = {}));
var ClientException = (function (_super) {
    __extends(ClientException, _super);
    function ClientException(opts) {
        var _this = _super.call(this, __assign({ name: "ClientException", $fault: "client" }, opts)) || this;
        _this.name = "ClientException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ClientException.prototype);
        return _this;
    }
    return ClientException;
}(__BaseException));
export { ClientException };
export var ManagedScalingStatus;
(function (ManagedScalingStatus) {
    ManagedScalingStatus["DISABLED"] = "DISABLED";
    ManagedScalingStatus["ENABLED"] = "ENABLED";
})(ManagedScalingStatus || (ManagedScalingStatus = {}));
export var ManagedTerminationProtection;
(function (ManagedTerminationProtection) {
    ManagedTerminationProtection["DISABLED"] = "DISABLED";
    ManagedTerminationProtection["ENABLED"] = "ENABLED";
})(ManagedTerminationProtection || (ManagedTerminationProtection = {}));
export var CapacityProviderStatus;
(function (CapacityProviderStatus) {
    CapacityProviderStatus["ACTIVE"] = "ACTIVE";
    CapacityProviderStatus["INACTIVE"] = "INACTIVE";
})(CapacityProviderStatus || (CapacityProviderStatus = {}));
export var CapacityProviderUpdateStatus;
(function (CapacityProviderUpdateStatus) {
    CapacityProviderUpdateStatus["DELETE_COMPLETE"] = "DELETE_COMPLETE";
    CapacityProviderUpdateStatus["DELETE_FAILED"] = "DELETE_FAILED";
    CapacityProviderUpdateStatus["DELETE_IN_PROGRESS"] = "DELETE_IN_PROGRESS";
    CapacityProviderUpdateStatus["UPDATE_COMPLETE"] = "UPDATE_COMPLETE";
    CapacityProviderUpdateStatus["UPDATE_FAILED"] = "UPDATE_FAILED";
    CapacityProviderUpdateStatus["UPDATE_IN_PROGRESS"] = "UPDATE_IN_PROGRESS";
})(CapacityProviderUpdateStatus || (CapacityProviderUpdateStatus = {}));
var InvalidParameterException = (function (_super) {
    __extends(InvalidParameterException, _super);
    function InvalidParameterException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidParameterException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidParameterException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidParameterException.prototype);
        return _this;
    }
    return InvalidParameterException;
}(__BaseException));
export { InvalidParameterException };
var LimitExceededException = (function (_super) {
    __extends(LimitExceededException, _super);
    function LimitExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "LimitExceededException", $fault: "client" }, opts)) || this;
        _this.name = "LimitExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, LimitExceededException.prototype);
        return _this;
    }
    return LimitExceededException;
}(__BaseException));
export { LimitExceededException };
var ServerException = (function (_super) {
    __extends(ServerException, _super);
    function ServerException(opts) {
        var _this = _super.call(this, __assign({ name: "ServerException", $fault: "server" }, opts)) || this;
        _this.name = "ServerException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, ServerException.prototype);
        return _this;
    }
    return ServerException;
}(__BaseException));
export { ServerException };
var UpdateInProgressException = (function (_super) {
    __extends(UpdateInProgressException, _super);
    function UpdateInProgressException(opts) {
        var _this = _super.call(this, __assign({ name: "UpdateInProgressException", $fault: "client" }, opts)) || this;
        _this.name = "UpdateInProgressException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, UpdateInProgressException.prototype);
        return _this;
    }
    return UpdateInProgressException;
}(__BaseException));
export { UpdateInProgressException };
export var ExecuteCommandLogging;
(function (ExecuteCommandLogging) {
    ExecuteCommandLogging["DEFAULT"] = "DEFAULT";
    ExecuteCommandLogging["NONE"] = "NONE";
    ExecuteCommandLogging["OVERRIDE"] = "OVERRIDE";
})(ExecuteCommandLogging || (ExecuteCommandLogging = {}));
export var ClusterSettingName;
(function (ClusterSettingName) {
    ClusterSettingName["CONTAINER_INSIGHTS"] = "containerInsights";
})(ClusterSettingName || (ClusterSettingName = {}));
var ClusterNotFoundException = (function (_super) {
    __extends(ClusterNotFoundException, _super);
    function ClusterNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "ClusterNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "ClusterNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ClusterNotFoundException.prototype);
        return _this;
    }
    return ClusterNotFoundException;
}(__BaseException));
export { ClusterNotFoundException };
export var DeploymentControllerType;
(function (DeploymentControllerType) {
    DeploymentControllerType["CODE_DEPLOY"] = "CODE_DEPLOY";
    DeploymentControllerType["ECS"] = "ECS";
    DeploymentControllerType["EXTERNAL"] = "EXTERNAL";
})(DeploymentControllerType || (DeploymentControllerType = {}));
export var LaunchType;
(function (LaunchType) {
    LaunchType["EC2"] = "EC2";
    LaunchType["EXTERNAL"] = "EXTERNAL";
    LaunchType["FARGATE"] = "FARGATE";
})(LaunchType || (LaunchType = {}));
export var AssignPublicIp;
(function (AssignPublicIp) {
    AssignPublicIp["DISABLED"] = "DISABLED";
    AssignPublicIp["ENABLED"] = "ENABLED";
})(AssignPublicIp || (AssignPublicIp = {}));
export var PlacementConstraintType;
(function (PlacementConstraintType) {
    PlacementConstraintType["DISTINCT_INSTANCE"] = "distinctInstance";
    PlacementConstraintType["MEMBER_OF"] = "memberOf";
})(PlacementConstraintType || (PlacementConstraintType = {}));
export var PlacementStrategyType;
(function (PlacementStrategyType) {
    PlacementStrategyType["BINPACK"] = "binpack";
    PlacementStrategyType["RANDOM"] = "random";
    PlacementStrategyType["SPREAD"] = "spread";
})(PlacementStrategyType || (PlacementStrategyType = {}));
export var PropagateTags;
(function (PropagateTags) {
    PropagateTags["NONE"] = "NONE";
    PropagateTags["SERVICE"] = "SERVICE";
    PropagateTags["TASK_DEFINITION"] = "TASK_DEFINITION";
})(PropagateTags || (PropagateTags = {}));
export var SchedulingStrategy;
(function (SchedulingStrategy) {
    SchedulingStrategy["DAEMON"] = "DAEMON";
    SchedulingStrategy["REPLICA"] = "REPLICA";
})(SchedulingStrategy || (SchedulingStrategy = {}));
export var DeploymentRolloutState;
(function (DeploymentRolloutState) {
    DeploymentRolloutState["COMPLETED"] = "COMPLETED";
    DeploymentRolloutState["FAILED"] = "FAILED";
    DeploymentRolloutState["IN_PROGRESS"] = "IN_PROGRESS";
})(DeploymentRolloutState || (DeploymentRolloutState = {}));
export var ScaleUnit;
(function (ScaleUnit) {
    ScaleUnit["PERCENT"] = "PERCENT";
})(ScaleUnit || (ScaleUnit = {}));
export var StabilityStatus;
(function (StabilityStatus) {
    StabilityStatus["STABILIZING"] = "STABILIZING";
    StabilityStatus["STEADY_STATE"] = "STEADY_STATE";
})(StabilityStatus || (StabilityStatus = {}));
var PlatformTaskDefinitionIncompatibilityException = (function (_super) {
    __extends(PlatformTaskDefinitionIncompatibilityException, _super);
    function PlatformTaskDefinitionIncompatibilityException(opts) {
        var _this = _super.call(this, __assign({ name: "PlatformTaskDefinitionIncompatibilityException", $fault: "client" }, opts)) || this;
        _this.name = "PlatformTaskDefinitionIncompatibilityException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, PlatformTaskDefinitionIncompatibilityException.prototype);
        return _this;
    }
    return PlatformTaskDefinitionIncompatibilityException;
}(__BaseException));
export { PlatformTaskDefinitionIncompatibilityException };
var PlatformUnknownException = (function (_super) {
    __extends(PlatformUnknownException, _super);
    function PlatformUnknownException(opts) {
        var _this = _super.call(this, __assign({ name: "PlatformUnknownException", $fault: "client" }, opts)) || this;
        _this.name = "PlatformUnknownException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, PlatformUnknownException.prototype);
        return _this;
    }
    return PlatformUnknownException;
}(__BaseException));
export { PlatformUnknownException };
var UnsupportedFeatureException = (function (_super) {
    __extends(UnsupportedFeatureException, _super);
    function UnsupportedFeatureException(opts) {
        var _this = _super.call(this, __assign({ name: "UnsupportedFeatureException", $fault: "client" }, opts)) || this;
        _this.name = "UnsupportedFeatureException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, UnsupportedFeatureException.prototype);
        return _this;
    }
    return UnsupportedFeatureException;
}(__BaseException));
export { UnsupportedFeatureException };
var ServiceNotActiveException = (function (_super) {
    __extends(ServiceNotActiveException, _super);
    function ServiceNotActiveException(opts) {
        var _this = _super.call(this, __assign({ name: "ServiceNotActiveException", $fault: "client" }, opts)) || this;
        _this.name = "ServiceNotActiveException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ServiceNotActiveException.prototype);
        return _this;
    }
    return ServiceNotActiveException;
}(__BaseException));
export { ServiceNotActiveException };
var ServiceNotFoundException = (function (_super) {
    __extends(ServiceNotFoundException, _super);
    function ServiceNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "ServiceNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "ServiceNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ServiceNotFoundException.prototype);
        return _this;
    }
    return ServiceNotFoundException;
}(__BaseException));
export { ServiceNotFoundException };
export var SettingName;
(function (SettingName) {
    SettingName["AWSVPC_TRUNKING"] = "awsvpcTrunking";
    SettingName["CONTAINER_INSIGHTS"] = "containerInsights";
    SettingName["CONTAINER_INSTANCE_LONG_ARN_FORMAT"] = "containerInstanceLongArnFormat";
    SettingName["SERVICE_LONG_ARN_FORMAT"] = "serviceLongArnFormat";
    SettingName["TASK_LONG_ARN_FORMAT"] = "taskLongArnFormat";
})(SettingName || (SettingName = {}));
export var TargetType;
(function (TargetType) {
    TargetType["CONTAINER_INSTANCE"] = "container-instance";
})(TargetType || (TargetType = {}));
var TargetNotFoundException = (function (_super) {
    __extends(TargetNotFoundException, _super);
    function TargetNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "TargetNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "TargetNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, TargetNotFoundException.prototype);
        return _this;
    }
    return TargetNotFoundException;
}(__BaseException));
export { TargetNotFoundException };
var ClusterContainsContainerInstancesException = (function (_super) {
    __extends(ClusterContainsContainerInstancesException, _super);
    function ClusterContainsContainerInstancesException(opts) {
        var _this = _super.call(this, __assign({ name: "ClusterContainsContainerInstancesException", $fault: "client" }, opts)) || this;
        _this.name = "ClusterContainsContainerInstancesException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ClusterContainsContainerInstancesException.prototype);
        return _this;
    }
    return ClusterContainsContainerInstancesException;
}(__BaseException));
export { ClusterContainsContainerInstancesException };
var ClusterContainsServicesException = (function (_super) {
    __extends(ClusterContainsServicesException, _super);
    function ClusterContainsServicesException(opts) {
        var _this = _super.call(this, __assign({ name: "ClusterContainsServicesException", $fault: "client" }, opts)) || this;
        _this.name = "ClusterContainsServicesException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ClusterContainsServicesException.prototype);
        return _this;
    }
    return ClusterContainsServicesException;
}(__BaseException));
export { ClusterContainsServicesException };
var ClusterContainsTasksException = (function (_super) {
    __extends(ClusterContainsTasksException, _super);
    function ClusterContainsTasksException(opts) {
        var _this = _super.call(this, __assign({ name: "ClusterContainsTasksException", $fault: "client" }, opts)) || this;
        _this.name = "ClusterContainsTasksException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ClusterContainsTasksException.prototype);
        return _this;
    }
    return ClusterContainsTasksException;
}(__BaseException));
export { ClusterContainsTasksException };
var TaskSetNotFoundException = (function (_super) {
    __extends(TaskSetNotFoundException, _super);
    function TaskSetNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "TaskSetNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "TaskSetNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, TaskSetNotFoundException.prototype);
        return _this;
    }
    return TaskSetNotFoundException;
}(__BaseException));
export { TaskSetNotFoundException };
export var InstanceHealthCheckState;
(function (InstanceHealthCheckState) {
    InstanceHealthCheckState["IMPAIRED"] = "IMPAIRED";
    InstanceHealthCheckState["INITIALIZING"] = "INITIALIZING";
    InstanceHealthCheckState["INSUFFICIENT_DATA"] = "INSUFFICIENT_DATA";
    InstanceHealthCheckState["OK"] = "OK";
})(InstanceHealthCheckState || (InstanceHealthCheckState = {}));
export var InstanceHealthCheckType;
(function (InstanceHealthCheckType) {
    InstanceHealthCheckType["CONTAINER_RUNTIME"] = "CONTAINER_RUNTIME";
})(InstanceHealthCheckType || (InstanceHealthCheckType = {}));
export var Compatibility;
(function (Compatibility) {
    Compatibility["EC2"] = "EC2";
    Compatibility["EXTERNAL"] = "EXTERNAL";
    Compatibility["FARGATE"] = "FARGATE";
})(Compatibility || (Compatibility = {}));
export var ContainerCondition;
(function (ContainerCondition) {
    ContainerCondition["COMPLETE"] = "COMPLETE";
    ContainerCondition["HEALTHY"] = "HEALTHY";
    ContainerCondition["START"] = "START";
    ContainerCondition["SUCCESS"] = "SUCCESS";
})(ContainerCondition || (ContainerCondition = {}));
export var EnvironmentFileType;
(function (EnvironmentFileType) {
    EnvironmentFileType["S3"] = "s3";
})(EnvironmentFileType || (EnvironmentFileType = {}));
export var FirelensConfigurationType;
(function (FirelensConfigurationType) {
    FirelensConfigurationType["FLUENTBIT"] = "fluentbit";
    FirelensConfigurationType["FLUENTD"] = "fluentd";
})(FirelensConfigurationType || (FirelensConfigurationType = {}));
export var DeviceCgroupPermission;
(function (DeviceCgroupPermission) {
    DeviceCgroupPermission["MKNOD"] = "mknod";
    DeviceCgroupPermission["READ"] = "read";
    DeviceCgroupPermission["WRITE"] = "write";
})(DeviceCgroupPermission || (DeviceCgroupPermission = {}));
export var LogDriver;
(function (LogDriver) {
    LogDriver["AWSFIRELENS"] = "awsfirelens";
    LogDriver["AWSLOGS"] = "awslogs";
    LogDriver["FLUENTD"] = "fluentd";
    LogDriver["GELF"] = "gelf";
    LogDriver["JOURNALD"] = "journald";
    LogDriver["JSON_FILE"] = "json-file";
    LogDriver["SPLUNK"] = "splunk";
    LogDriver["SYSLOG"] = "syslog";
})(LogDriver || (LogDriver = {}));
export var TransportProtocol;
(function (TransportProtocol) {
    TransportProtocol["TCP"] = "tcp";
    TransportProtocol["UDP"] = "udp";
})(TransportProtocol || (TransportProtocol = {}));
export var ResourceType;
(function (ResourceType) {
    ResourceType["GPU"] = "GPU";
    ResourceType["INFERENCE_ACCELERATOR"] = "InferenceAccelerator";
})(ResourceType || (ResourceType = {}));
export var UlimitName;
(function (UlimitName) {
    UlimitName["CORE"] = "core";
    UlimitName["CPU"] = "cpu";
    UlimitName["DATA"] = "data";
    UlimitName["FSIZE"] = "fsize";
    UlimitName["LOCKS"] = "locks";
    UlimitName["MEMLOCK"] = "memlock";
    UlimitName["MSGQUEUE"] = "msgqueue";
    UlimitName["NICE"] = "nice";
    UlimitName["NOFILE"] = "nofile";
    UlimitName["NPROC"] = "nproc";
    UlimitName["RSS"] = "rss";
    UlimitName["RTPRIO"] = "rtprio";
    UlimitName["RTTIME"] = "rttime";
    UlimitName["SIGPENDING"] = "sigpending";
    UlimitName["STACK"] = "stack";
})(UlimitName || (UlimitName = {}));
export var IpcMode;
(function (IpcMode) {
    IpcMode["HOST"] = "host";
    IpcMode["NONE"] = "none";
    IpcMode["TASK"] = "task";
})(IpcMode || (IpcMode = {}));
export var NetworkMode;
(function (NetworkMode) {
    NetworkMode["AWSVPC"] = "awsvpc";
    NetworkMode["BRIDGE"] = "bridge";
    NetworkMode["HOST"] = "host";
    NetworkMode["NONE"] = "none";
})(NetworkMode || (NetworkMode = {}));
export var PidMode;
(function (PidMode) {
    PidMode["HOST"] = "host";
    PidMode["TASK"] = "task";
})(PidMode || (PidMode = {}));
export var TaskDefinitionPlacementConstraintType;
(function (TaskDefinitionPlacementConstraintType) {
    TaskDefinitionPlacementConstraintType["MEMBER_OF"] = "memberOf";
})(TaskDefinitionPlacementConstraintType || (TaskDefinitionPlacementConstraintType = {}));
export var ProxyConfigurationType;
(function (ProxyConfigurationType) {
    ProxyConfigurationType["APPMESH"] = "APPMESH";
})(ProxyConfigurationType || (ProxyConfigurationType = {}));
export var CPUArchitecture;
(function (CPUArchitecture) {
    CPUArchitecture["ARM64"] = "ARM64";
    CPUArchitecture["X86_64"] = "X86_64";
})(CPUArchitecture || (CPUArchitecture = {}));
export var OSFamily;
(function (OSFamily) {
    OSFamily["LINUX"] = "LINUX";
    OSFamily["WINDOWS_SERVER_2004_CORE"] = "WINDOWS_SERVER_2004_CORE";
    OSFamily["WINDOWS_SERVER_2016_FULL"] = "WINDOWS_SERVER_2016_FULL";
    OSFamily["WINDOWS_SERVER_2019_CORE"] = "WINDOWS_SERVER_2019_CORE";
    OSFamily["WINDOWS_SERVER_2019_FULL"] = "WINDOWS_SERVER_2019_FULL";
    OSFamily["WINDOWS_SERVER_2022_CORE"] = "WINDOWS_SERVER_2022_CORE";
    OSFamily["WINDOWS_SERVER_2022_FULL"] = "WINDOWS_SERVER_2022_FULL";
    OSFamily["WINDOWS_SERVER_20H2_CORE"] = "WINDOWS_SERVER_20H2_CORE";
})(OSFamily || (OSFamily = {}));
export var TaskDefinitionStatus;
(function (TaskDefinitionStatus) {
    TaskDefinitionStatus["ACTIVE"] = "ACTIVE";
    TaskDefinitionStatus["INACTIVE"] = "INACTIVE";
})(TaskDefinitionStatus || (TaskDefinitionStatus = {}));
export var Scope;
(function (Scope) {
    Scope["SHARED"] = "shared";
    Scope["TASK"] = "task";
})(Scope || (Scope = {}));
export var EFSAuthorizationConfigIAM;
(function (EFSAuthorizationConfigIAM) {
    EFSAuthorizationConfigIAM["DISABLED"] = "DISABLED";
    EFSAuthorizationConfigIAM["ENABLED"] = "ENABLED";
})(EFSAuthorizationConfigIAM || (EFSAuthorizationConfigIAM = {}));
export var EFSTransitEncryption;
(function (EFSTransitEncryption) {
    EFSTransitEncryption["DISABLED"] = "DISABLED";
    EFSTransitEncryption["ENABLED"] = "ENABLED";
})(EFSTransitEncryption || (EFSTransitEncryption = {}));
export var CapacityProviderField;
(function (CapacityProviderField) {
    CapacityProviderField["TAGS"] = "TAGS";
})(CapacityProviderField || (CapacityProviderField = {}));
export var ClusterField;
(function (ClusterField) {
    ClusterField["ATTACHMENTS"] = "ATTACHMENTS";
    ClusterField["CONFIGURATIONS"] = "CONFIGURATIONS";
    ClusterField["SETTINGS"] = "SETTINGS";
    ClusterField["STATISTICS"] = "STATISTICS";
    ClusterField["TAGS"] = "TAGS";
})(ClusterField || (ClusterField = {}));
export var ContainerInstanceField;
(function (ContainerInstanceField) {
    ContainerInstanceField["CONTAINER_INSTANCE_HEALTH"] = "CONTAINER_INSTANCE_HEALTH";
    ContainerInstanceField["TAGS"] = "TAGS";
})(ContainerInstanceField || (ContainerInstanceField = {}));
export var ServiceField;
(function (ServiceField) {
    ServiceField["TAGS"] = "TAGS";
})(ServiceField || (ServiceField = {}));
export var TaskDefinitionField;
(function (TaskDefinitionField) {
    TaskDefinitionField["TAGS"] = "TAGS";
})(TaskDefinitionField || (TaskDefinitionField = {}));
export var TaskField;
(function (TaskField) {
    TaskField["TAGS"] = "TAGS";
})(TaskField || (TaskField = {}));
export var Connectivity;
(function (Connectivity) {
    Connectivity["CONNECTED"] = "CONNECTED";
    Connectivity["DISCONNECTED"] = "DISCONNECTED";
})(Connectivity || (Connectivity = {}));
export var HealthStatus;
(function (HealthStatus) {
    HealthStatus["HEALTHY"] = "HEALTHY";
    HealthStatus["UNHEALTHY"] = "UNHEALTHY";
    HealthStatus["UNKNOWN"] = "UNKNOWN";
})(HealthStatus || (HealthStatus = {}));
export var ManagedAgentName;
(function (ManagedAgentName) {
    ManagedAgentName["ExecuteCommandAgent"] = "ExecuteCommandAgent";
})(ManagedAgentName || (ManagedAgentName = {}));
export var TaskStopCode;
(function (TaskStopCode) {
    TaskStopCode["ESSENTIAL_CONTAINER_EXITED"] = "EssentialContainerExited";
    TaskStopCode["TASK_FAILED_TO_START"] = "TaskFailedToStart";
    TaskStopCode["USER_INITIATED"] = "UserInitiated";
})(TaskStopCode || (TaskStopCode = {}));
export var TaskSetField;
(function (TaskSetField) {
    TaskSetField["TAGS"] = "TAGS";
})(TaskSetField || (TaskSetField = {}));
var TargetNotConnectedException = (function (_super) {
    __extends(TargetNotConnectedException, _super);
    function TargetNotConnectedException(opts) {
        var _this = _super.call(this, __assign({ name: "TargetNotConnectedException", $fault: "client" }, opts)) || this;
        _this.name = "TargetNotConnectedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, TargetNotConnectedException.prototype);
        return _this;
    }
    return TargetNotConnectedException;
}(__BaseException));
export { TargetNotConnectedException };
export var ContainerInstanceStatus;
(function (ContainerInstanceStatus) {
    ContainerInstanceStatus["ACTIVE"] = "ACTIVE";
    ContainerInstanceStatus["DEREGISTERING"] = "DEREGISTERING";
    ContainerInstanceStatus["DRAINING"] = "DRAINING";
    ContainerInstanceStatus["REGISTERING"] = "REGISTERING";
    ContainerInstanceStatus["REGISTRATION_FAILED"] = "REGISTRATION_FAILED";
})(ContainerInstanceStatus || (ContainerInstanceStatus = {}));
export var TaskDefinitionFamilyStatus;
(function (TaskDefinitionFamilyStatus) {
    TaskDefinitionFamilyStatus["ACTIVE"] = "ACTIVE";
    TaskDefinitionFamilyStatus["ALL"] = "ALL";
    TaskDefinitionFamilyStatus["INACTIVE"] = "INACTIVE";
})(TaskDefinitionFamilyStatus || (TaskDefinitionFamilyStatus = {}));
export var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
export var DesiredStatus;
(function (DesiredStatus) {
    DesiredStatus["PENDING"] = "PENDING";
    DesiredStatus["RUNNING"] = "RUNNING";
    DesiredStatus["STOPPED"] = "STOPPED";
})(DesiredStatus || (DesiredStatus = {}));
var AttributeLimitExceededException = (function (_super) {
    __extends(AttributeLimitExceededException, _super);
    function AttributeLimitExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "AttributeLimitExceededException", $fault: "client" }, opts)) || this;
        _this.name = "AttributeLimitExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AttributeLimitExceededException.prototype);
        return _this;
    }
    return AttributeLimitExceededException;
}(__BaseException));
export { AttributeLimitExceededException };
var ResourceInUseException = (function (_super) {
    __extends(ResourceInUseException, _super);
    function ResourceInUseException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceInUseException", $fault: "client" }, opts)) || this;
        _this.name = "ResourceInUseException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceInUseException.prototype);
        return _this;
    }
    return ResourceInUseException;
}(__BaseException));
export { ResourceInUseException };
export var PlatformDeviceType;
(function (PlatformDeviceType) {
    PlatformDeviceType["GPU"] = "GPU";
})(PlatformDeviceType || (PlatformDeviceType = {}));
var BlockedException = (function (_super) {
    __extends(BlockedException, _super);
    function BlockedException(opts) {
        var _this = _super.call(this, __assign({ name: "BlockedException", $fault: "client" }, opts)) || this;
        _this.name = "BlockedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, BlockedException.prototype);
        return _this;
    }
    return BlockedException;
}(__BaseException));
export { BlockedException };
var ResourceNotFoundException = (function (_super) {
    __extends(ResourceNotFoundException, _super);
    function ResourceNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "ResourceNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceNotFoundException.prototype);
        return _this;
    }
    return ResourceNotFoundException;
}(__BaseException));
export { ResourceNotFoundException };
var MissingVersionException = (function (_super) {
    __extends(MissingVersionException, _super);
    function MissingVersionException(opts) {
        var _this = _super.call(this, __assign({ name: "MissingVersionException", $fault: "client" }, opts)) || this;
        _this.name = "MissingVersionException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, MissingVersionException.prototype);
        return _this;
    }
    return MissingVersionException;
}(__BaseException));
export { MissingVersionException };
var NoUpdateAvailableException = (function (_super) {
    __extends(NoUpdateAvailableException, _super);
    function NoUpdateAvailableException(opts) {
        var _this = _super.call(this, __assign({ name: "NoUpdateAvailableException", $fault: "client" }, opts)) || this;
        _this.name = "NoUpdateAvailableException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, NoUpdateAvailableException.prototype);
        return _this;
    }
    return NoUpdateAvailableException;
}(__BaseException));
export { NoUpdateAvailableException };
export var ManagedScalingFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AutoScalingGroupProviderFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TagFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateCapacityProviderRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CapacityProviderFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateCapacityProviderResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ExecuteCommandLogConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ExecuteCommandConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ClusterConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CapacityProviderStrategyItemFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ClusterSettingFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateClusterRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var KeyValuePairFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AttachmentFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ClusterFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateClusterResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeploymentCircuitBreakerFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeploymentConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeploymentControllerFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var LoadBalancerFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AwsVpcConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var NetworkConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PlacementConstraintFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PlacementStrategyFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ServiceRegistryFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateServiceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeploymentFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ServiceEventFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ScaleFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TaskSetFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ServiceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateServiceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateTaskSetRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateTaskSetResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteAccountSettingRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SettingFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteAccountSettingResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AttributeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteCapacityProviderRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteCapacityProviderResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteClusterRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteClusterResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteServiceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteServiceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteTaskSetRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteTaskSetResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeregisterContainerInstanceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var InstanceHealthCheckResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContainerInstanceHealthStatusFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ResourceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var VersionInfoFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContainerInstanceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeregisterContainerInstanceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeregisterTaskDefinitionRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContainerDependencyFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EnvironmentFileFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var HostEntryFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FirelensConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var HealthCheckFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var KernelCapabilitiesFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeviceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TmpfsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var LinuxParametersFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SecretFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var LogConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var MountPointFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PortMappingFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RepositoryCredentialsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ResourceRequirementFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SystemControlFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UlimitFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var VolumeFromFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContainerDefinitionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EphemeralStorageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var InferenceAcceleratorFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TaskDefinitionPlacementConstraintFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ProxyConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RuntimePlatformFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DockerVolumeConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EFSAuthorizationConfigFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EFSVolumeConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FSxWindowsFileServerAuthorizationConfigFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FSxWindowsFileServerVolumeConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var HostVolumePropertiesFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var VolumeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TaskDefinitionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeregisterTaskDefinitionResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeCapacityProvidersRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FailureFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeCapacityProvidersResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeClustersRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeClustersResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeContainerInstancesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeContainerInstancesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeServicesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeServicesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeTaskDefinitionRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeTaskDefinitionResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeTasksRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ManagedAgentFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var NetworkBindingFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var NetworkInterfaceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContainerFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContainerOverrideFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var InferenceAcceleratorOverrideFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TaskOverrideFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TaskFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeTasksResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeTaskSetsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DescribeTaskSetsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DiscoverPollEndpointRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DiscoverPollEndpointResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ExecuteCommandRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SessionFilterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.tokenValue && { tokenValue: SENSITIVE_STRING }))); };
export var ExecuteCommandResponseFilterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.session && { session: SessionFilterSensitiveLog(obj.session) }))); };
export var ListAccountSettingsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListAccountSettingsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListClustersRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListClustersResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListContainerInstancesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListContainerInstancesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListServicesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListServicesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTagsForResourceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTagsForResourceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTaskDefinitionFamiliesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTaskDefinitionFamiliesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTaskDefinitionsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTaskDefinitionsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTasksRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTasksResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountSettingRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountSettingResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountSettingDefaultRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountSettingDefaultResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutClusterCapacityProvidersRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutClusterCapacityProvidersResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PlatformDeviceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RegisterContainerInstanceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RegisterContainerInstanceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RegisterTaskDefinitionRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RegisterTaskDefinitionResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RunTaskRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RunTaskResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var StartTaskRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var StartTaskResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var StopTaskRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var StopTaskResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AttachmentStateChangeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SubmitAttachmentStateChangesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SubmitAttachmentStateChangesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SubmitContainerStateChangeRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SubmitContainerStateChangeResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContainerStateChangeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ManagedAgentStateChangeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SubmitTaskStateChangeRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SubmitTaskStateChangeResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TagResourceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TagResourceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UntagResourceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UntagResourceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AutoScalingGroupProviderUpdateFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateCapacityProviderRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateCapacityProviderResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateClusterRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateClusterResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateClusterSettingsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateClusterSettingsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateContainerAgentRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateContainerAgentResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateContainerInstancesStateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateContainerInstancesStateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateServiceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateServiceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateServicePrimaryTaskSetRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateServicePrimaryTaskSetResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateTaskSetRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateTaskSetResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
